clippy_lints/src/methods/filter_map_identity.rs

```diff
@@ -1,22 +1,34 @@
 use clippy_utils::diagnostics::span_lint_and_sugg;
-use clippy_utils::{is_expr_untyped_identity_function, is_trait_method};
+use clippy_utils::{is_expr_identity_function, is_expr_untyped_identity_function, is_trait_method};
 use rustc_errors::Applicability;
 use rustc_hir as hir;
 use rustc_lint::LateContext;
 use rustc_span::{sym, Span};
 
 use super::FILTER_MAP_IDENTITY;
 
+fn is_identity(cx: &LateContext<'_>, expr: &hir::Expr<'_>) -> Option<Applicability> {
+    if is_expr_untyped_identity_function(cx, expr) {
+        return Some(Applicability::MachineApplicable);
+    }
+    if is_expr_identity_function(cx, expr) {
+        return Some(Applicability::MaybeIncorrect);
```

Comment:
This should be `Unspecified` (or `HasPlaceholders`).

`MaybeIncorrect` suggestions should still compile.

Reply:
I've interpreted "valid Rust code" as it parsing correctly, everything else can fail I'm pretty sure

Reply:
I remember @xFrednet commenting about it [here](https://github.com/rust-lang/rust-clippy/pull/11919#discussion_r1415930314)

But oh well, the fact that we have a different understanding is all the more reason that the documentation/naming for `Applicability` could be better

Reply:
rustfix somewhere has an issue to rename these variants to make the difference clearer. I think it's mostly stuck on finding better names: 

https://github.com/rust-lang/cargo/issues/13028